plugin/dnstap/dnstapio/io.go

```diff
@@ -2,67 +2,99 @@ package dnstapio
 
 import (
 	"log"
+	"net"
+	"time"
 
 	tap "github.com/dnstap/golang-dnstap"
+	fs "github.com/farsightsec/golang-framestream"
 	"github.com/golang/protobuf/proto"
 )
 
-// DnstapIO wraps the dnstap I/O routine.
-type DnstapIO struct {
-	protocol Protocol
-	queue    chan tap.Dnstap
-	stop     chan bool
+const (
+	tcpTimeout   = 4 * time.Second
+	flushTimeout = 1 * time.Second
+	queueSize    = 1000
+)
+
+type dnstapIO struct {
+	enc   *fs.Encoder
+	conn  net.Conn
+	queue chan tap.Dnstap
 }
 
-// Protocol is either `out.TCP` or `out.Socket`.
-type Protocol interface {
-	// Write takes a single frame at once.
-	Write([]byte) (int, error)
+// New dnstap I/O routine.
```

Comment:
New returns a new and initialized DnstapIO

Reply:
Fixed